src/App.tsx

```diff
@@ -1,223 +1,176 @@
+  const [isAdding, setIsAdding] = useState(false);
+
+  const filteredTodo = [...todos].filter(todo => {
+    switch (sortType) {
+      case SortType.Active:
+        return !todo.completed;
+      case SortType.Completed:
+        return todo.completed;
+      default:
+        return todo;
+    }
+  });
+
   useEffect(() => {
-    // focus the element with `ref={newTodoField}`
     if (newTodoField.current) {
       newTodoField.current.focus();
     }
   }, []);
 
+  useEffect(() => {
+    const getTodosFromServer = async (userId: number) => {
+      try {
+        const receivedTodos = await getTodos(userId);
+
+        setTodos(receivedTodos);
+      } catch (errorMessage) {
+        setError(`${errorMessage}`);
+      }
+    };
+
+    if (!user) {
+      return;
+    }
+
+    getTodosFromServer(user.id);
+  }, []);
+
+  const getValue = (element: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(element.target.value);
+  };
+
+  const addNewTodo = useCallback(async (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!title.trim() || !user) {
+      setError(ErrorType.ErrorTitle);
+
+      return;
+    }
+
+    setIsAdding(true);
+
+    try {
+      const postTodo = await createTodo(title, user.id);
+
+      setTodos((prevTodos) => [...prevTodos, postTodo]);
+    } catch {
+      setError(ErrorType.NotAdd);
+    }
+
+    setIsAdding(false);
+    setTitle('');
+  }, [title, user]);
+
+  const removeTodo = useCallback(async (TodoId: number) => {
+    setSelectedId([TodoId]);
+
+    try {
+      await deleteTodos(TodoId);
+
+      setTodos((prevTodos) => prevTodos.filter(({ id }) => id !== TodoId));
+    } catch {
+      setError(ErrorType.NotDelete);
+    }
+  }, [todos, error]);
+
+  const completedTodos = todos.filter(({ completed }) => completed);
+
+  const clearCompletedTodos = useCallback(() => {
+    setSelectedId([...completedTodos].map(({ id }) => id));
+
+    Promise.all(completedTodos.map(({ id }) => removeTodo(id)))
+      .then(() => setTodos((prevTodos) => prevTodos
+        .filter(({ completed }) => !completed)))
+      .catch(() => {
+        setError(ErrorType.NotDelete);
+        setSelectedId([]);
+      });
+  }, [todos, selectedId, error]);
+
   return (
     <div className="todoapp">
       <h1 className="todoapp__title">todos</h1>
-
       <div className="todoapp__content">
         <header className="todoapp__header">
           <button
             data-cy="ToggleAllButton"
             type="button"
             className="todoapp__toggle-all active"
           />
-
-          <form>
+          <form onSubmit={addNewTodo}>
             <input
               data-cy="NewTodoField"
               type="text"
               ref={newTodoField}
               className="todoapp__new-todo"
               placeholder="What needs to be done?"
+              value={title}
+              onChange={getValue}
             />
           </form>
         </header>
```

Comment:
I could suggest u to create a separate component here too, but it's up to u

Reply:
++